### DISCUSSION TITLE
Inappropriate Exception Message for Long Blocking Connect Function

### DISCUSSION THREAD

**Discussion by AuroraTea** [miguelgrinberg/python-socketio#1419] [General]
If the blocking connect function takes too long, it will be considered as a connection dropout, and the Exception received on the client side is "One or more namespaces failed to connect." This might be inappropriate as it can easily mislead beginners.

**Comment by miguelgrinberg** [comment]
From the client side a connection taking too long cannot be diagnosed accurately. The error could be `ConnectionTimeout` or something like that, but I view this as being potentially more misleading than the neutral "failed to connect".

How were you thinking this can be improved?
